sui_programmability/examples/nft_standard/sources/example_nft.move

```diff
@@ -0,0 +1,75 @@
+module nft_standard::example_nft {
+
+    /// The NFT itself.
+    struct ExampleNFT has key, store {
+        id: UID,
+        /// Proof that this NFT belongs to `collection`
+        collection: CollectionProof,
+        // metadata understood by the "display" standards goes here.
+        name: String,
+        url: Url,
+        // .. more fields
+        // ..., or alternatively, we could put the "standard" metadata in a common structure like
+        // https://github.com/Origin-Byte/nft-protocol/blob/main/sources/nft/std_nft.move#L22 and
+        // ask folks to use this
+    }
+
+    /// The mint policy. This is a shared object that allows someone to mint an NFT by paying the required fee.
+    /// In all likelihood, there would be libraries for common drop schemes, auctions, etc. that NFT creators can leverage.
+    /// But a creator can also define a totally custom policy using arbitrary Move code (e.g., only someone that owns a
+    /// JellyNFT can mint a PeanutButterNFT, ...)
+    struct MintPolicy has key {
+        id: UID,
+        mint_cap: MintCap,
+        /// Price of the NFT, in SUI
+        price: u64,
+    }
+
+    /// The royalty policy. This is a shared object that allows a buyer to complete a sale by
+    /// complying with it.
+    /// In all likelihood, there would be libraries for common policies that NFT creators can
```

Comment:
3. We appreciate the advantage of the proposed `RoyaltyReceipt` in terms of a single definition of collection's `RoyaltyPolicy`. One defines a policy once and henceforth all TSC refer to that implementation. However, aside from having a new OB contract (or auction, etc) per collection, we struggle to see how to implement it due to point 1 (https://github.com/MystenLabs/sui/pull/4887/files#r984856303). We will further prompt how we could enable arbitrary royalty strategies better.

3a. Custom royalty policies, such as the one mentioned in the example, will require an incompatible `foo_nft::buy` function signatures